deadbelly/breathEZ
# Bje/feature/app component

### What does this PR do? 
This branch fleshes out `App.js` and `apiCalls.js`. It includes logic for retrieving the data we want from the api and a skeleton for how to render it. 

### Is this a feature or Fix/Refactor?  
Feature

### Problems Encountered & Resulting Solutions  
I hit a big blocker when trying to fetch all available cities in the US on page load. This triggered too many api calls at once, and no throttling method that I tried worked. Instead I refactored so that the user must first select a state, and then select a city. This logic should maybe move to our `<Search />` component in a refactor, but I wanted to ensure that it's working before moving on. 

### Where should the reviewer start?
Look at `App.js` and then `apiCalls.js` for the relevant logic.

### How should this be tested?
Boot up the app and navigate to the components panel of your developer tools. Notice the list of state names. Pay attention to the order of the hooks. You should be able to change each hook string manually, and if you use valid data the next hook will automatically update. 

### Steps To Take
Separating out some concerns is our next priority. Some of this logic should live in a controlled form in `<Search />`. We also need to update the return statement now that more components exist. 

### This PR is related to which issues:
close #6
